src/team3player/MinerRobot.java

```diff
@@ -3,15 +3,42 @@
 class MinerRobot extends RobotFramework {
     MapLocation hqLoc;
 
+    int OurRobot[];
+    int BuildTarget = 5;
+    int GatherTarget = 1;
+    int RobotPoint = 0;
+
     MinerRobot(RobotController rc_) {
         //super(rc_) calls the constructor of the parent class which just saves rc
         //the parent class also has the old utility functions like tryMove which need rc
         super(rc_);
         //on robot creation/start up code goes here
+
+        // Avi Code Starts
+
+        RobotPoint += 1 ;
+
         System.out.println("Miner:" + rc.getID() + " initialization");
 
-        // Avi Code here
+        OurRobot[RobotPoint] = rc.getID();
+
+        if ( RobotPoint % BuildTarget == 0 ) {
+            buildMode(); }
+
+        else if ( RobotPoint % GatherTarget == 0 ) {
+            gatherMode(); }
+
+        else {
+            buildMode();
+        }
+
+
+        if ( RobotPoint % 6 == 0 && BuildTarget > 1) {
+            BuildTarget -= 1;
+            GatherTarget += 1;
+        }
```

Comment:
The constructor only gets called once, so having something like var +=1 just means var = 1. The bots do not have shared memory, you can not coordinate bots with shared variables. You would see that all your robots go into buildmode, but there is a null pointer exception while the game runs.